### PR TITLE
[5.8] Document new features in Homestead

### PR DESCRIPTION
**Pull request by deviantintegral**
For details, see https://github.com/laravel/homestead/issues/1071. A new version hasn't been built or tagged yet but should be soon.

I'm a little unsure of how to document the new mDNS support. For our teams, _everything_ is set up on a per-site basis, so defaulting to mDNS is the simplest setup. It's especially useful if you're using DHCP for VM IPs instead of static addresses. But, we're not typically using Laravel in our projects either, so I'm not sure what the broader community defaults are.

/cc @svpernova09

### CODE REVIEW

**Comment by taylorotwell** [comment]
Thanks for PR. Just a heads up gonna take me a bit to review this because of various capitalization / formatting issues that are inconsistent with the rest of the documentation.

**Comment by deviantintegral** [comment]
I got a first pass done. Are there docs somewhere on how to build the docs site? Closest I've found is the README here and https://laravel.com/docs/5.8/readme. That'd make it easier for me to catch any glaring omissions.

File: homestead.md [hunk 1]
```diff
@@ -514,6 +524,72 @@ Homestead uses the Nginx web server by default. However, it can install Apache i
...
+
+<a name="debugging-and-profiling"></a>
+## Debugging & Profiling
+
+<a name="debugging-web-requests"></a>
+### Debugging Web Requests With Xdebug
+
+Homestead includes out-of-the-box support for step debugging with [Xdebug](https://xdebug.org). For example, you can load a web page from a browser, and PHP will connect back out to your IDE to allow inspection and modification of the running code.
+
+To enable debugging, run:
```
Comment (taylorotwell) [inline homestead.md]: Where?

File: homestead.md [hunk 2]
```diff
@@ -514,6 +524,72 @@ Homestead uses the Nginx web server by default. However, it can install Apache i
 
 Homestead includes the Postfix mail transfer agent, which is listening on port `1025` by default. So, you may instruct your application to use the `smtp` mail driver on `localhost` port `1025`. Then, all sent mail will be handled by Postfix and caught by Mailhog. To view your sent emails, open [http://localhost:8025](http://localhost:8025) in your web browser.
 
+<a name="database-snapshots"></a>
+### Database Snapshots
```
Comment (taylorotwell) [inline homestead.md]: I feel like this section never gives me a step-by-step guide of how to actually use the feature.

File: homestead.md [hunk 3]
```diff
@@ -514,6 +524,72 @@ Homestead uses the Nginx web server by default. However, it can install Apache i
...
+Loading Xdebug by itself, even when not actively debugging, can significantly slow PHP. To disable Xdebug, run `sudo phpdismod xdebug` and restart the FPM service again.
+
+While Xdebug is included for all PHP versions, note that early minor releases of PHP often have some incompatibilities with Xdebug immediately after release. If PHP segfaults or throws Zend errors, try switching to the previous stable PHP release.
+
+<a name="debugging-cli-scripts"></a>
+### Debugging CLI scripts
+
+To debug a PHP CLI program, use the `xphp` shell alias:
+
+    $ xphp path/to/script
```
Comment (taylorotwell) [inline homestead.md]: Where?

File: homestead.md [hunk 4]
```diff
@@ -514,6 +524,72 @@ Homestead uses the Nginx web server by default. However, it can install Apache i
...
+
+When debugging functional tests that make requests to the web server, it is often easier to autostart debugging rather than modifying tests to pass through a custom header or cookie to trigger debugging. To force Xdebug to start, modify `/etc/php/7.#/fpm/conf.d/20-xdebug.ini` (replacing the PHP version) and add:
+
```
Comment (taylorotwell) [inline homestead.md]: Where?

File: homestead.md [hunk 5]
```diff
@@ -514,6 +524,72 @@ Homestead uses the Nginx web server by default. However, it can install Apache i
...
+    ; If Homestead.yml contains a different subnet for the ip setting, this may be different.
+    xdebug.remote_host = 192.168.10.1
+    xdebug.remote_autostart = 1
+
+### Profiling PHP performance with xhgui
```
Comment (taylorotwell) [inline homestead.md]: Please use title case.

**Comment by deviantintegral** [comment]
Thanks @taylorotwell , this is ready for another review.

File: homestead.md (hunk 3 above)
Comment (svpernova09) [reply homestead.md]: We should probably change this to `xphp path/to/script.php` or similar to give more context

File: homestead.md (hunk 2 above)
Comment (svpernova09) [reply homestead.md]: I'm happy to take a shot at this once we have the new base box out and I can play around with it.

File: homestead.md (hunk 1 above)
Comment (svpernova09) [reply homestead.md]: These 2 commands can be run from anywhere.

**Pull request merged**